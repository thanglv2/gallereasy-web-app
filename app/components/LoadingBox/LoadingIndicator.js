import styled, { keyframes } from 'styled-components';

const spin = keyframes`
  0% { transform: rotate(0deg); }
  100% { transform: rotate(360deg); }
`;

const LoadingIndicator = styled.div`
  margin: 50px auto;
  border: 8px solid ${props => props.theme.darkgrey};
  border-radius: 50%;
  border-top: 8px solid ${props => props.theme.dodgerblue};
  width: 60px;
  height: 60px;
  animation: ${spin} 2s linear infinite;
`;

const Wrapper = styled.div`
  width: 100%;
  align-items: center;
`;

export { LoadingIndicator, Wrapper };