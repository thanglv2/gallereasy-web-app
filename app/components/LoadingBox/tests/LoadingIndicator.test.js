import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import { LoadingIndicator } from '../LoadingIndicator';

describe('<LoadingIndicator />', () => {
  it('render without error', () => {
    const theme = {
      darkgrey: 'darkgrey',
      dodgerblue: 'dodgerblue'
    }
    const tree = renderer.create(<LoadingIndicator theme={theme} />).toJSON()
    expect(tree).toMatchSnapshot()
  });
});