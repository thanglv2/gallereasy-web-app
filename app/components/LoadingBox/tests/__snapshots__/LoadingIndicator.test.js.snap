// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingIndicator /> render without error 1`] = `
.c0 {
  margin: 0 auto;
  border: 8px solid darkgrey;
  border-radius: 50%;
  border-top: 8px solid dodgerblue;
  width: 60px;
  height: 60px;
  -webkit-animation: iECmZH 2s linear infinite;
  animation: iECmZH 2s linear infinite;
}

<div
  className="c0"
/>
`;
